src/composite-view.js

```diff
@@ -15,164 +14,48 @@ const ClassOptions = [
+    this._getChildViewContainer();
   },
 
-  renderChildren() {
-    if (this._isRendered || this._isRendering) {
-      CollectionView.prototype._renderChildren.call(this);
-    }
-  },
-
-  // You might need to override this if you've overridden attachHtml
-  attachBuffer(compositeView, buffer) {
-    const $container = this.getChildViewContainer(compositeView);
-    this.Dom.appendContents($container[0], buffer, {_$el: $container});
-  },
-
-  // Internal method. Append a view to the end of the $el.
-  // Overidden from CollectionView to ensure view is appended to
-  // childViewContainer
-  _insertAfter(childView) {
-    const $container = this.getChildViewContainer(this, childView);
-    this.Dom.appendContents($container[0], childView.el, {_$el: $container, _$contents: childView.$el});
-  },
-
-  // Internal method. Append reordered childView'.
-  // Overidden from CollectionView to ensure reordered views
-  // are appended to childViewContainer
-  _appendReorderedChildren(children) {
-    const $container = this.getChildViewContainer(this);
-    this.Dom.appendContents($container[0], children, {_$el: $container});
-  },
-
-  // Internal method to ensure an `$childViewContainer` exists, for the
-  // `attachHtml` method to use.
-  getChildViewContainer(containerView, childView) {
-    if (!!containerView.$childViewContainer) {
-      return containerView.$childViewContainer;
-    }
+  _getChildViewContainer() {
+    const childViewContainer = _.result(this, 'childViewContainer');
+    this.$container = childViewContainer ? this.$(childViewContainer) : this.$el;
 
-    let container;
-    const childViewContainer = containerView.childViewContainer;
-    if (childViewContainer) {
-
-      const selector = _.result(containerView, 'childViewContainer');
-
-      if (selector.charAt(0) === '@' && containerView.ui) {
-        container = containerView.ui[selector.substr(4)];
-      } else {
-        container = this.$(selector);
-      }
-
-      if (container.length <= 0) {
-        throw new MarionetteError({
-          name: 'ChildViewContainerMissingError',
-          message: `The specified "childViewContainer" was not found: ${containerView.childViewContainer}`
-        });
-      }
-
-    } else {
-      container = containerView.$el;
-    }
-
-    containerView.$childViewContainer = container;
-    return container;
-  },
-
-  // Internal method to reset the `$childViewContainer` on render
-  resetChildViewContainer() {
-    if (this.$childViewContainer) {
-      this.$childViewContainer = undefined;
+    if (this.$container.length <= 0) {
+      throw new MarionetteError({
+        name: 'ChildViewContainerMissingError',
+        message: `The specified "childViewContainer" was not found: ${childViewContainer}`
+      });
     }
   }
 });
 
-// To prevent duplication but allow the best View organization
-// Certain View methods are mixed directly into the deprecated CompositeView
+// TODO: Extract to a TemplateRenderMixin
```

Comment:
If we go this way we should officially move these functions to a mixin that is also used by view.. I _kinda_ like that idea anyway